### PR TITLE
[process-agent] Separate queue/forwarder for network connections

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

Create a separate queue and corresponding forwarder for network connections payloads.

### Motivation

Ensure submission errors/retries for connections do not affect other payload types (processes/containers and RT payloads).

### Describe how to test/QA your changes

Regular QA of network connections functionality + validation of queue stats in the status.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/process-agent/collector.go [hunk 1]
```diff
@@ -360,14 +368,16 @@ func (l *Collector) run(exit chan struct{}) error {
...
+					eventQueueSize:        l.eventResults.Len(),
+					podQueueSize:          l.podResults.Len(),
+					processQueueBytes:     l.processResults.Weight(),
+					rtProcessQueueBytes:   l.rtProcessResults.Weight(),
+					connectionsQueueBytes: l.connectionsResults.Weight(),
+					eventQueueBytes:       l.eventResults.Weight(),
+					podQueueBytes:         l.podResults.Weight(),
 				})
 			case <-queueLogTicker.C:
 				l.logQueuesSize()
```
Comment (robertjli) [inline cmd/process-agent/collector.go:383]: Does this function need to be updated as well? Not sure how we decide which queues to log or not

File: cmd/process-agent/info.go [hunk 2]
```diff
@@ -162,25 +166,29 @@ func updateProcContainerCount(msgs []model.MessageBody) {
...
+	processQueueBytes     int64
+	rtProcessQueueBytes   int64
+	connectionsQueueBytes int64
+	eventQueueBytes       int64
+	podQueueBytes         int64
 }
 
 func updateQueueStats(stats *queueStats) {
 	infoMutex.Lock()
 	defer infoMutex.Unlock()
```
Comment (robertjli) [inline cmd/process-agent/info.go:183]: Do we actually need this mutex? I could see a case where the length and memory footprint of a queue don't correlate if the reads for these values are threaded in with these writes, how much do we care if that happens?

Comment (xornivore) [reply cmd/process-agent/info.go:183]: Discussed offline, will bulk change in a subsequent PR.

File: cmd/process-agent/collector.go (hunk 1 above)
Comment (xornivore) [reply cmd/process-agent/collector.go:383]: Good catch, I missed this spot.

File: cmd/process-agent/collector.go [hunk 3]
```diff
@@ -127,6 +130,9 @@ func NewCollectorWithChecks(cfg *config.AgentConfig, checks []checks.Check, runR
 	rtProcessResults := api.NewWeightedQueue(rtQueueSize, int64(queueBytes))
 	log.Debugf("Creating rt process check queue with max_size=%d and max_weight=%d", rtProcessResults.MaxSize(), rtProcessResults.MaxWeight())
 
+	connectionsResults := api.NewWeightedQueue(queueSize, int64(queueBytes))
```
[the remark this answers is not in the document]
Comment (xornivore) [reply cmd/process-agent/collector.go:133]: This one is tricky without making the queue settings equal - you can have either of these features on or off, and both on. There is a potential for exceeding memory in a container, but it poses a lesser risk than lowering the default setting.

[the remark this answers is not in the document]
Comment (robertjli) [reply cmd/process-agent/collector.go:133]: Discussed it offline, I think it's fine but something to keep in mind. If customers notice a significant increase based on their configuration, we can direct them to lower the queue size and weight.

File: pkg/process/util/status.go [hunk 4]
```diff
@@ -66,29 +66,31 @@ type MemInfo struct {
...
+	UptimeNano            float64             `json:"uptime_nano"`
+	MemStats              MemInfo             `json:"memstats"`
+	Version               InfoVersion         `json:"version"`
+	DockerSocket          string              `json:"docker_socket"`
+	LastCollectTime       string              `json:"last_collect_time"`
+	ProcessCount          int                 `json:"process_count"`
+	ContainerCount        int                 `json:"container_count"`
+	ProcessQueueSize      int                 `json:"process_queue_size"`
+	RTProcessQueueSize    int                 `json:"rtprocess_queue_size"`
+	ConnectionsQueueSize  int                 `json:"connections_queue_size"`
```
Comment (mbotarro) [inline pkg/process/util/status.go:80]: We may want to add the new fields to the templated used to render the status command:
https://github.com/DataDog/datadog-agent/blob/1d15a135240a86d31a48e86e0e983c73b1cbf24c/pkg/status/templates/process-agent.tmpl#L39-L54

**Review by mbotarro — APPROVED** [review]

File: pkg/process/util/status.go (hunk 4 above)
Comment (xornivore) [reply pkg/process/util/status.go:80]: Updated, thank you!

**Review by robertjli — APPROVED** [review]
LGTM, thanks for the updates.

In the future, I'm curious what we think about customizing the output based on the enabled checks?

**Pull request merged**